### PR TITLE
fix dataset plot and improve the plotting

### PR DESCRIPTION
**Pull request by Louis-Dupont**
- Pascal VOC fixed (it used to use a deprecated params that actually wasnt supported anymore for plotting)
- Making `visualize_image` a public method with default params.
- Moving from "custom" plotting to using the `visualize_image`

### CODE REVIEW

File: src/super_gradients/training/datasets/detection_datasets/detection_dataset.py [hunk 1]
```diff
@@ -529,34 +539,62 @@ def plot(self, max_samples_per_plot: int = 16, n_plots: int = 1, plot_transforme
...
-                boxes = sample["target"][:, 0:4]
+                image = image.astype(np.uint8)
+                image = cv2.cvtColor(image, cv2.COLOR_BGR2RGB)  # Detection dataset works with BGR images, so we have to convert to RGB
+
+                # Convert to XYXY_LABEL format
+                targets_format_converter = ConcatenatedTensorFormatConverter(input_format=input_format, output_format=XYXY_LABEL, image_shape=image.shape)
+                targets_xyxy_label = targets_format_converter(targets)
+
+                # PLOT TARGETS
+                color_mapping = generate_color_mapping(len(self.classes))
```
Comment (BloodAxe) [inline src/super_gradients/training/datasets/detection_datasets/detection_dataset.py:576]: We have DetectionVisualization that contains many similar routines we use here and I wonder maybe we can concentrate all drawing logic in that class and call DetectionVisualization._visualize_image (make sense to make method public)? WDYT?

Comment (Louis-Dupont) [reply src/super_gradients/training/datasets/detection_datasets/detection_dataset.py]: I pushed it here [2b2dbc7](https://github.com/Deci-AI/super-gradients/pull/1686/commits/2b2dbc784016fb255e8c9bf602b8c3fe43e99318)

**Comment by BloodAxe** [comment]
Overall LGTM, just ensure that you've updated public API change in notebooks (model_export does use DetectionVisualization)

File: src/super_gradients/training/utils/detection_utils.py [hunk 2]
```diff
@@ -423,16 +425,20 @@ def draw_box_title(
         :param y2: Y coordinate of the bottom right corner of the bounding box
         :param class_id: A corresponding class id
         :param pred_conf: Class confidence score (optional)
-        :param is_target: Indicate if the bounding box is a ground-truth box or not
-
+        :param bbox_prefix: Prefix to add to the title of the bounding boxes
+        :param is_target: Not Supported Anymore. Please use  `pred_conf` and `bbox_prefix` instead.
         """
+        if is_target:
```
Comment (BloodAxe) [inline src/super_gradients/training/utils/detection_utils.py:431]: Although this method is public I think there are really low chances someone is using it aside. So it should be safe to just remove the argument completely

Comment (Louis-Dupont) [reply src/super_gradients/training/utils/detection_utils.py]: Removed in [efd3579](https://github.com/Deci-AI/super-gradients/pull/1686/commits/efd357918383b7a750dc690d31cbf535d2089b2a)

**Review by BloodAxe — APPROVED** [review]
LGTM

**Comment by ani-mal** [comment]
When will this change be released. I am running into an error because the `box[1]` value is `int64` and not `int` for `draw_box_tiles(...)`, this PR seems to address this issue:

Missing `int` casting here on version 3.5.0:
 
https://github.com/Deci-AI/super-gradients/blob/4d68589d5e846a1183a0bcbbd77deb2722fdc925/src/super_gradients/training/utils/detection_utils.py#L459-L462

**Pull request merged**